src/meltano/core/venv_service.py

```diff
@@ -454,24 +559,165 @@ async def extract_stderr(proc: Process) -> str | None:  # pragma: no cover
             return (await proc.stdout.read()).decode("utf-8", errors="replace")
 
         try:
-            return await exec_async(
-                str(self.exec_path("python")),
-                "-m",
-                "pip",
-                "install",
-                "--log",
-                str(self.pip_log_path),
-                *pip_install_args,
+            return await self.install_pip_args(
+                pip_install_args,
                 extract_stderr=extract_stderr,
                 env=env,
             )
-        except AsyncSubprocessError as err:
-            logger.info(
-                "Logged pip install output to %s",  # noqa: WPS323
-                self.pip_log_path,
-            )
-            raise AsyncSubprocessError(
-                f"Failed to install plugin '{self.name}'.",  # noqa: EM102
-                err.process,
-                stderr=await err.stderr,
-            ) from err
+        except AsyncSubprocessError as err:  # noqa: WPS329
+            raise await self.handle_installation_error(err) from err
+
+
+class UvVenvService(VenvService):
+    """Manages virtual environments using `uv`."""
+
+    @staticmethod
+    def _find_uv() -> str:  # noqa: WPS605
+        """Find the `uv` executable.
+
+        Tries to import the `uv` package and use its `find_uv_bin` function to find the
+        `uv` executable. If that fails, falls back to using the `uv` executable on the
+        system PATH. If it can't be found on the PATH, returns `"uv"`.
+
+        Adapted from https://github.com/wntrblm/nox/blob/55c7eaf2eb03feb4a4b79e74966c542b75d61401/nox/virtualenv.py#L42-L54.
+
+        Copyright 2016 Alethea Katherine Flowers
+
+        Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+           http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+
+        Returns:
+            A string representing the path to the `uv` executable.
+        """
+        with contextlib.suppress(ImportError, FileNotFoundError):
+            from uv import find_uv_bin
+
+            return find_uv_bin()
+
+        # Fall back to PATH.
+        uv = shutil.which("uv")  # pragma: no cover
+        return uv or "uv"  # pragma: no cover
```

Comment:
Maybe we should fail if `uv` is not in the `PATH` after trying to find it in the Python venv?

Reply:
I think it makes sense to fail if `uv` is not in `$PATH`. I'm not sure how we could expect it to run if we can't find it, since when we call `subprocess` later it'll search for the executable on `$PATH` too unless we provide a path to an executable.